ros_ign_bridge/src/service_factory.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SERVICE_FACTORY_HPP_
+#define SERVICE_FACTORY_HPP_
+
+#include <ignition/transport/Node.hh>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "ros_ign_bridge/convert_decl.hpp"
+
+#include "service_factory_interface.hpp"
+
+namespace ros_ign_bridge
+{
+
+template<typename RosResT>
+bool
+send_response_on_error(RosResT & ros_response);
+
+template<typename RosServiceT, typename IgnRequestT, typename IgnReplyT>
```

Comment:
```suggestion
template<typename RosServiceT, typename IgnRequestT, typename IgnResponseT>
```

"response" is used throughout Ignition Transport's codebase

Reply:
Actually, [Reply is used](https://github.com/ignitionrobotics/ign-transport/blob/e02bd8b99bdcaff9bae401d791236317dd27c6d3/include/ignition/transport/Node.hh#L371).
In `ign-gazebo` though, there are more instances of "response" than "reply".

If `Response` is more common for ignition developers, I will update it here.
It also matches the ROS 2 naming.